DentOS_Framework/DentOsTestbedLib/src/dent_os_testbed/lib/traffic/ixnetwork/ixnetwork_ixia_client_impl.py

```diff
@@ -663,3 +666,30 @@ def run_send_arp(self, device, command, *argv, **kwarg):
         return err, [{"success": msg["arg2"],
                       "port": msg["port"],
                       "src_ip": msg["src_ip"]} for msg in res]
+
+    def run_update_l1_config(self, device, command, *argv, **kwarg):
+        speeds_map = {"1000": "speed1000",
+                      "100": "speed100fd",
+                      "10000": "speed10g",
+                      "2500": "speed2.5g",
+                      "5000": "speed5g"}
+
+        if not IxnetworkIxiaClientImpl.ixnet:
+            return 0, "Ixia not connected"
+        if command == "update_l1_config":
+            vports = IxnetworkIxiaClientImpl.ixnet.Vport.find()
+            ports = kwarg['params'][0].get("ixia_ports", None)
+            speed = kwarg['params'][0].get("speed", None)
+            speed_to_ixia = None if speed is None else speeds_map[str(speed)]
+            autoneg = kwarg['params'][0].get("autoneg", True)
+            names = [vport.Name for vport in vports]
+            for port in ports:
+                if port in names:
```

Comment:
```suggestion
                if port not in names:
                    continue
```

Reply:
no difference as for me from `if port in names`

Reply:
There is a difference. You will have less indentation:
```python
for port in ports:
    if port in names:
        required_ixia_port = vports[names.index(port)]
        card = required_ixia_port.L1Config.NovusTenGigLan
        card.update(AutoNegotiate=autoneg, Speed=speed_to_ixia)
```
vs
```python
for port in ports:
    if port not in names:
        continue
    required_ixia_port = vports[names.index(port)]
    card = required_ixia_port.L1Config.NovusTenGigLan
    card.update(AutoNegotiate=autoneg, Speed=speed_to_ixia)
```

Reply:
Still not resolved

Reply:
Right here(line 676)
` if len(ports) < 1:
                raise ValueError(" IXIA ports not provided")`